diesel/src/associations/belongs_to.rs

```diff
@@ -27,8 +27,9 @@ impl<Parent, Child, Iter> GroupedBy<Parent> for Iter where
         let id_indices: HashMap<_, _> = parents.iter().enumerate().map(|(i, u)| (u.id(), i)).collect();
         let mut result = parents.iter().map(|_| Vec::new()).collect::<Vec<_>>();
         for child in self {
-            let index = id_indices[child.foreign_key()];
-            result[index].push(child);
+            if let Some(index) = child.foreign_key().map(|i| id_indices[i]) {
+                result[index].push(child);
+            }
         }
```

Comment:
It could probably be changed to

```rust
for (index, child) in self.into_iter().filter_map(|child| child.foreign_key().map(|k| (id_indices[k], child))
```

I can't turn the whole thing into a functional statement, as there's no method out there for this. The equivalent in Ruby would be something along the lines of `group_by { |child| id_indices[foreign_key] }`, but neither Rust nor itertools provide that method (itertools has a method with that name that does something extremely different)